redash/settings/__init__.py

```diff
@@ -48,9 +49,77 @@ def all_settings():
 SCHEMAS_REFRESH_SCHEDULE = int(os.environ.get("REDASH_SCHEMAS_REFRESH_SCHEDULE", 30))
 
 AUTH_TYPE = os.environ.get("REDASH_AUTH_TYPE", "api_key")
-ENFORCE_HTTPS = parse_boolean(os.environ.get("REDASH_ENFORCE_HTTPS", "false"))
 INVITATION_TOKEN_MAX_AGE = int(os.environ.get("REDASH_INVITATION_TOKEN_MAX_AGE", 60 * 60 * 24 * 7))
 
+# Whether and how to redirect non-HTTP requests to HTTPS. Disabled by default.
+ENFORCE_HTTPS = parse_boolean(os.environ.get("REDASH_ENFORCE_HTTPS", "false"))
+ENFORCE_HTTPS_PERMANENT = parse_boolean(
+    os.environ.get("REDASH_ENFORCE_HTTPS_PERMANENT", "false"))
+# Whether file downloads are enforced or not.
+ENFORCE_FILE_SAVE = parse_boolean(
+    os.environ.get("REDASH_ENFORCE_FILE_SAVE", "true"))
+
+COOKIE_SECRET = os.environ.get(
+    "REDASH_COOKIE_SECRET", "c292a0a3aa32397cdb050e233733900f")
+# Whether the session cookie is set to secure.
+SESSION_COOKIE_SECURE = parse_boolean(
+    os.environ.get("REDASH_SESSION_COOKIE_SECURE") or str(ENFORCE_HTTPS))
+# Whether the session cookie is set HttpOnly.
+SESSION_COOKIE_HTTPONLY = parse_boolean(
+    os.environ.get("REDASH_SESSION_COOKIE_HTTPONLY", "true"))
+
+# Doesn't set X-Frame-Options by default since it's highly dependent
+# on the specific deployment.
+# See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options
+# for more information.
+FRAME_OPTIONS = os.environ.get("REDASH_FRAME_OPTIONS", None)
+FRAME_OPTIONS_ALLOW_FROM = os.environ.get(
+    "REDASH_FRAME_OPTIONS_ALLOW_FROM", "")
+
+# Whether and how to send Strict-Transport-Security response headers.
+# See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+# for more information.
+HSTS_ENABLED = parse_boolean(os.environ.get("REDASH_HSTS_ENABLED", "false"))
+HSTS_PRELOAD = parse_boolean(os.environ.get("REDASH_HSTS_PRELOAD", "false"))
+HSTS_MAX_AGE = int(
+    os.environ.get("REDASH_HSTS_MAX_AGE", talisman.ONE_YEAR_IN_SECS))
+HSTS_INCLUDE_SUBDOMAINS = parse_boolean(
+    os.environ.get("REDASH_HSTS_INCLUDE_SUBDOMAINS", "false")
+)
+
+# Whether and how to send Content-Security-Policy response headers.
+# See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
+# for more information.
+CONTENT_SECURITY_POLICY = os.environ.get(
```

Comment:
`os.environ.get` returns a string, while the default value is an object. You probably want json.loads/dumps here.

Reply:
Ah yeah, so the values for policy-style config variables (CONTENT_SECURITY_POLICY and FEATURE_POLICY) can either be full blown dicts or strings that are parsed automatically by Talisman: https://github.com/GoogleCloudPlatform/flask-talisman/blob/3c1a0d3555b16e918cbdc9c0c769f37a0da24970/flask_talisman/talisman_test.py#L150-L168

That's why it's okay to define the values as dicts here and not have to use json for the env vars. If we were to set the CONTENT_SECURITY_POLICY or FEATURE_POLICY env vars, we'd have to use the string version basically (e.g. `CONTENT_SECURITY_POLICY="default-src 'self'"`)

Would you prefer json instead?

Reply:
In this case, let's add a link to the documentation on what formats are accepted here. Because people usually look at the default value to understand what's expected.

Reply:
Good point!